rskj-core/src/main/java/co/rsk/rpc/json/JsonResponseSizeLimiter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.rpc.json;
+
+import co.rsk.rpc.exception.JsonRpcResponseLimitError;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class JsonResponseSizeLimiter {
+    private static final int BRACKETS_SIZE_IN_BYTES = 2;
+    private static final int COMMA_SIZE = 1;
+    private static final int JSON_FIELD_MISSING_SYMBOLS = 3;
+
+    private int acc = 0;
+    private final int max;
+    private final JsonNode root;
+
+    private JsonResponseSizeLimiter(int max, JsonNode root) {
+        this.max = max;
+        this.root = root;
+    }
+
+    public static int getSizeInBytesWithLimit(JsonNode node, int limit) {
+        if (node == null) {
+            return 0;
+        }
+        JsonResponseSizeLimiter limiter = new JsonResponseSizeLimiter(limit, node);
+        return limiter.getJsonLength();
+    }
+
+    private int getJsonLength() {
+        addLength(root);
+        return acc;
+    }
+
+    private void addLength(JsonNode jsonNode) {
+        if (jsonNode.isArray()) {
+            handleArray((ArrayNode) jsonNode);
+        } else if (jsonNode.isValueNode()) {
+            handleValueNode(jsonNode);
+        } else {
+            handleObject(jsonNode);
+        }
+    }
+
+    private void handleValueNode(JsonNode node) {
+        sumAndCheck(node.toString().getBytes().length);
```

Comment:
```suggestion
        sumAndCheck(node.asText().length());
```
`JsonNode.toString()` involves extra memory allocation by using `ObjectWriter`, I'd rather use `asText()` one. Also `.getBytes()` allocates extra copy of it, probably `String.length()` should be sufficient, wdyt?

Reply:
asText() doesn't have the same result as toString(),  using it will require extra checks. 
Using `String.length()` instead of `getBytes().length` will work for ASCII characters which will be the most common so it can work.